synthesizer/process/src/lib.rs

```diff
@@ -66,30 +66,61 @@ use synthesizer_program::{
 };
 use synthesizer_snark::{ProvingKey, UniversalSRS, VerifyingKey};
 
-use aleo_std::prelude::{finish, lap, timer};
-use indexmap::IndexMap;
-use parking_lot::RwLock;
-use std::{collections::HashMap, sync::Arc};
+use aleo_std::{
+    StorageMode,
+    prelude::{finish, lap, timer},
+};
+use lru::LruCache;
+use parking_lot::{Mutex, RwLock};
+use std::{collections::HashMap, num::NonZeroUsize, sync::Arc};
+use tracing::debug;
 
 #[cfg(feature = "aleo-cli")]
 use colored::Colorize;
 
+#[cfg(not(feature = "rocks"))]
+use ledger_store::helpers::memory::ConsensusMemory;
+#[cfg(feature = "rocks")]
+use ledger_store::helpers::rocksdb::ConsensusDB;
+
+#[cfg(feature = "rocks")]
+#[derive(Clone)]
+pub struct Process<N: Network> {
+    /// The universal SRS.
+    universal_srs: Arc<UniversalSRS<N>>,
+    /// The Stack for credits.aleo
+    credits: Option<Arc<Stack<N>>>,
+    /// The mapping of program IDs to stacks.
+    stacks: Arc<Mutex<LruCache<ProgramID<N>, Arc<Stack<N>>>>>,
+    /// The storage.
+    store: Option<ConsensusStore<N, ConsensusDB<N>>>,
```

Comment:
perhaps feature-gating would work on just the `store` member? i.e.
```
    /// The storage.
    #[cfg(feature = "rocks")]
    store: Option<ConsensusStore<N, ConsensusDB<N>>>,
    #[cfg(not(feature = "rocks"))]
    store: Option<ConsensusStore<N, ConsensusMemory<N>>>,
```

Reply:
b1e2a0a8c